src/components/BookPage/index.js

```diff
@@ -47,12 +47,17 @@ function BookPage() {
         console.log(info.data.reviews.length);
       })
       .catch((error) => console.error("Error fetching data", error));
-  }, [id]);
+  }
+  useEffect(() => {
+    fetchBookData()
+  }, [id])
```

Comment:
Now you can just pass fetchBookData directly into useffect

`useEffect(fetchBookData, [id])`